developer_guides/add_new_arch.rst

```diff
@@ -0,0 +1,34 @@
+.. _add_new_arch:
+
+Adding a new DSP architecture to SOF
+====================================
+
+This is not yet a guide for architecure porting, but in general are two ways to 
+add support for new DSP architectures to SOF.
+
+#. Write a new Hardware Abstraction Layer (HAL) for your DSP.
+
+#. Use an existing RTOS that supports your DSP architecture as a HAL for SOF.
```

Comment:
I'd use a bullet list here instead of a numbered list...

Reply:
@anton-intel the purpose of this commit is to re-organize place where content exist in chapters tree - not to refactor the content itself.